include/PieceFactory.h

```diff
@@ -0,0 +1,28 @@
+#include <memory>
```

Comment:
You have included unnecessary headers such as <memory> and the headers for all the piece types. If those headers are not required for the declaration of the PieceFactory class, you can remove them to reduce unnecessary dependencies.

Reply:
Done.